### DISCUSSION TITLE
new dialect - HANA express docker image for integration tests in CI

### DISCUSSION THREAD

**Discussion by dazhuangc** [sequelize/sequelize#17132] [Help & Questions] [answered]
Hi,

I would like to contribute by adding HANA dialect. As we discussed in https://github.com/sequelize/sequelize/discussions/16982, we need to make sure that we can run CI, including integration tests, on HANA dialects. 
There is a HANA express docker image https://hub.docker.com/r/saplabs/hanaexpress, which can be used for the integration tests.
But I am not familiar with the Sequelize CI. Could you please share some information about how the CI is built?
And if I am supposed to add the docker image in the CI, or you can add it?
If I should add it, could you please share some guidelines (Maybe there's some additional configuration file in the repository that needs to updated)?

Thanks.

**Comment by WikiRik** [comment, marked as answer]
We're using GitHub Actions for our CI, mainly this file; https://github.com/sequelize/sequelize/blob/main/.github/workflows/ci.yml
You can add `hana` to `test-oldest-latest`, which will then try to start the servers with `yarn start-hana-oldest` or `yarn start-hana-latest` which is defined in https://github.com/sequelize/sequelize/blob/main/package.json
That links back to [the dev folder](https://github.com/sequelize/sequelize/tree/main/dev) where you can set start.sh scripts to build up the database. Test configuration is done here; https://github.com/sequelize/sequelize/blob/main/packages/core/test/config/config.ts but you'll need to implement the code first (or at least a boilerplate version) before tests will be able to run. More detailed documentation on that part will follow for v8, so you will have to do some investigation on your own in the meantime.

**Reply by dazhuangc** [reply]
I will try with the steps.

Thanks!
